rscripts/YAL5-single-codons.R

```diff
@@ -0,0 +1,1108 @@
+                                                 filter_for_frame, 
+                                                 filtering_frame, snapdisp)
+
+  # Add codon identities
+  transcript_tibbles <- left_join(total_codon_pos_counts, yeast_codon_pos_i200,
+                                  by = c("PosCodon", "Gene"), keep = FALSE,
+                                  copy = TRUE)
+
+  # turn into a tibble
+  transcript_gene_poscodon_frame <- tibble(
+    Gene = transcript_tibbles$Gene,
+    PosCodon = transcript_tibbles$PosCodon,
+    Count = transcript_tibbles$Count,
+    Codon = transcript_tibbles$Codon
+  )
+
+  return(transcript_gene_poscodon_frame)
+}
+
+##TEST: Expect to produce a tibble with each position in the CDS having the
+# correct codon beside it.
+
+#TEST: AddCodonNamesToCodonPosCounts(): returns a tibble.
+#TEST: AddCodonNamesToCodonPosCounts(): the tibble has 4 columns.
+#TEST: AddCodonNamesToCodonPosCounts(): the column names are %in% c("Gene",
+# "PosCodon", "Count", "Codon").
+#TEST: AddCodonNamesToCodonPosCounts(): number of observations in the output
+# tibble = sum of CDS (codon co-ordinates) for all genes in gene_names.
+#TEST: AddCodonNamesToCodonPosCounts(): the unique gene names in column "Gene"
+# match the genes in gene_names
+# (unique(total_codon_pos_counts$Gene) = gene_names) = TRUE.
+# 
+#Example: using Mok-tinysim data, the following tibble is returned 
+# # A tibble: 9 x 4
+# Gene  PosCodon Count Codon
+# <chr>    <dbl> <dbl> <chr>
+#   1 MAT          1     0 ATG  
+# 2 MAT          2     2 GCC  
+# 3 MAT          3     2 ACA  
+# 4 MAT          4     0 TGA  
+# 5 MIKE         1     0 ATG  
+# 6 MIKE         2     1 ATC  
+# 7 MIKE         3     0 AAG  
+# 8 MIKE         4     0 GAG  
+# 9 MIKE         5     0 TAA  
+
+
+###  Slice out window around interesting features ###
+
+
+
+#' ExpandFeatureRegionAllGenes(): this function takes a slice out of
+#' transcript_gene_poscodon_frame, centered on the position of
+#' the feature of interest
+#' It then sets the position of the feature of interest to 0, and changes the
+#' positions of adjacent codons to be relative to the feature of interest
+#' This is done for all occurrences of the feature of interest on all genes in
+#' the sample provided.
+#'
+#' @param gene_names The list of all the genes in the sample 
+#' @param gene The gene from gene_names to get read lengths for.
+#' @param dataset The name of dataset stored in .h5 file.
+#' @param hd_file The path to the .h5 hdf5 file holding read data for all genes,
+#' created from BAM files for dataset samples.
+#' @param gff_df The gff file in a dataframe format
+#' @param min_read_length An integer, the minimum read length in H5 output;
+#' Default = 10 (set in generate_stats_figs.R from yaml)
+#' @param filter_for_frame Decides which method of asite assignment to use
+#' @param filtering_frame The frame from which to select reads from.
+#' @param snapdisp An integer for additional displacement in the snapping
+#'
```

Comment:
`yeast_codon_pos_i200` (the tsv file), `feature_of_interest`, `expand_width` and `remove_overhang` are listed as inputs for the function but are missing from the params list